app/src/main/java/explorer/city/com/cityexplorer/view/adapter/SearchAdapter.kt

```diff
@@ -26,7 +26,10 @@ class SearchAdapter(listener: OnItemClickListener<String>) : RecyclerView.Adapte
 
     override fun onBindViewHolder(vh: SearchViewHolder, position: Int) {
         val city = results[position]
-        vh.setSearchResult(city)
+        val imageUrl = city.embeddedSearch.cityInfo.embeddedUrbanArea?.urbanArea?.embeddedImage?.photos?.photos?.get(0)?.image?.web
+        val name = city.embeddedSearch.cityInfo.name
+        vh.setCityName(name)
+                .setCityImage(imageUrl)
```

Comment:
Here's an opportunity to just pass in the city. It's down to preference but there is logic in just letting the viewholder decide how to render its contents rather than having the adapter decide it.